h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -1189,56 +1222,50 @@ public void learn() {
-    Log.info(userFeedback.toString("User Feedback for AutoML Run " + this._key));
-    Log.info();
-
-
-
-    // TODO: Look into adding optionally adding these back in
-    // We should not spend time computing train/valid leaderboards until we are ready to expose them to the user
-    // Commenting out for now
-
-    /*
-    // Use a throwaway AutoML instance so the "New leader" message doesn't pollute our feedback
-    AutoML dummyAutoML = new AutoML();
-    UserFeedback dummyUF = new UserFeedback(dummyAutoML);
-    dummyAutoML.userFeedback = dummyUF;
-
-    Leaderboard trainingLeaderboard = Leaderboard.getOrMakeLeaderboard(projectName() + "_training", dummyUF, this.trainingFrame);
-    trainingLeaderboard.addModels(this.leaderboard().getModelKeys());
-    Log.info(trainingLeaderboard.toTwoDimTable("TRAINING FRAME Leaderboard for project " + projectName(), true).toString());
-    Log.info();
-
-    // Use a throwawayUserFeedback instance so the "New leader" message doesn't pollute our feedback
-    Leaderboard validationLeaderboard = Leaderboard.getOrMakeLeaderboard(projectName() + "_validation", dummyUF, this.validationFrame);
-    validationLeaderboard.addModels(this.leaderboard().getModelKeys());
-    Log.info(validationLeaderboard.toTwoDimTable("VALIDATION FRAME Leaderboard for project " + projectName(), true).toString());
-    Log.info();
-    */
-
-    Log.info(leaderboard().toTwoDimTable("Leaderboard for project " + projectName(), true).toString());
+    Log.info(userFeedback.toString("User Feedback for AutoML Run " + this._key + ":"));
+    for (UserFeedbackEvent event : userFeedback.feedbackEvents)
+      Log.info(event);
+
+    if (0 < this.leaderboard().getModelKeys().length) {
+      // Use a throwaway AutoML instance so the "New leader" message doesn't pollute our feedback
+      AutoML dummyAutoML = new AutoML();
+      UserFeedback dummyUF = new UserFeedback(dummyAutoML);
+      dummyAutoML.userFeedback = dummyUF;
+
+      Leaderboard trainingLeaderboard = Leaderboard.getOrMakeLeaderboard(projectName() + "_training", dummyUF, this.trainingFrame);
+      trainingLeaderboard.addModels(this.leaderboard().getModelKeys());
+      Log.info(trainingLeaderboard.toTwoDimTable("TRAINING FRAME Leaderboard for project " + projectName(), true).toString());
+      Log.info();
+
+      // Use a throwawayUserFeedback instance so the "New leader" message doesn't pollute our feedback
+      Leaderboard validationLeaderboard = Leaderboard.getOrMakeLeaderboard(projectName() + "_validation", dummyUF, this.validationFrame);
+      validationLeaderboard.addModels(this.leaderboard().getModelKeys());
+      Log.info(validationLeaderboard.toTwoDimTable("VALIDATION FRAME Leaderboard for project " + projectName(), true).toString());
+      Log.info();
+
+      Log.info(leaderboard().toTwoDimTable("Leaderboard for project " + projectName(), true).toString());
+    }
 
```

Comment:
The reason I commented this out is because it's expensive to score the leaderboard and until we are ready to return training and validation leaderboards to the user (rather than just print to logfile), we should not compute this.  Let's leave this commented out until we need it in the future.

Reply:
Fine.